website/static/js/share/index.js

```diff
@@ -48,16 +49,42 @@ ShareApp.controller = function() {
     var self = this;
 
     self.vm = new ShareApp.ViewModel(self.vm);
-    self.statsController = new Stats.controller(self.vm);
-    self.resultsController = new Results.controller(self.vm);
-    self.searchBarController = new SearchBar.controller(self.vm);
+
+    m.request({
+        method: 'get',
+        background: false,
+        url: '/api/v1/share/providers/'
+    }).then(function(data) {
+        self.vm.ProviderMap = data.providerMap;
+
+        self.sideBarController = new SideBar.controller(self.vm);
+        self.statsController = new Stats.controller(self.vm);
+        self.resultsController = new Results.controller(self.vm);
+        self.searchBarController = new SearchBar.controller(self.vm);
+
+    });
+
+    self.renderSort = function(){
```

Comment:
Could you move this into an external function and have it take the controller as an argument.